MultiSelectComboBox/Sdl.MultiSelectComboBox/Themes/Generic/MultiSelectComboBox.cs

```diff
@@ -167,7 +170,7 @@ private CollectionViewSource ItemsCollectionViewSource
                     if (EnableGrouping)
                     {
                         // check that the items are groupable before adding a default group definition
-                        if (ItemsCollectionViewSource.GroupDescriptions.Count == 0 && ItemsSource.Count > 0 && ItemsSource[0] is IItemGroupAware)
+                        if (ItemsCollectionViewSource.GroupDescriptions.Count == 0 && (ItemsSource.GetType().IsGenericType && typeof(IItemGroupAware).IsAssignableFrom(ItemsSource.GetType().GetGenericArguments()[0]) || ItemsSource.Count > 0 && ItemsSource[0] is IItemGroupAware))
```

Comment:
Can you improve the readability for this if and explain through code what you are trying to achieve, please?

Reply:
@ituriSdl I cleaned the code but logic is the same.
I want to handle case when we use suggestion provider:
- ItemsSource is ObservableCollection<LanguageItem> but contains zero elements
- new code detects that generic argument of the ItemsSource type implements IItemGroupAware and enable grouping
- ItemsSource is filled using suggestion provider after control is loaded

If you remove that code then control will not detect that we use groupable type.
There is a workaround to add to collection one dummy new LanguageItem() and this item will be removed by the first call to suggestion provider. But I want to avoid that requirement so I added this change.